### ISSUE TITLE
Enable starting WSL with genie installed cross-distro, i.e. full-pledge systemd system as a VSCode server.

### ISSUE THREAD

**Issue by PavelSosin-320** [microsoft/vscode-remote-release#4381]
Start WSL distro with all services that I need and want to use to be installed as systemd units as "remote" VSCode server. In other words, after Genie installation using wsl -d myDistro -u myUser genie -s
with httpd, podman, buildah, ... ready to serve me.

**Comment by aeschli** [comment]
Sorry, I'm not familiar with genie. How would the command line look that runs WSL with genie *and* the vscode-server (`sh wslServer.sh`)?

The Remote WSL extension already has the capability to run a user provided setup script: https://code.visualstudio.com/docs/remote/wsl#_advanced-environment-setup-script
Is that useful?

**Comment by PavelSosin-320** [comment]
@aeschli Genie makes Linux distro systemd based. Every service can be run using systemd unit file in the predefined sequence. Every socket can be allocated using systemd unit file. All unit files are executed automatically upon distro "boot": wsl -d distro -u root genie -s. 
Every user login via genie -l creates an isolated systemd session. systemd can manage cgroups.
Most of the modern Linux distro providers already adopted systemd and corresponding unit files are created during package installation. Units are manageable using systemctl.
In other words, no external script is needed to allocate a socket and run the server and stop it upon systemctl poweroff.
Using systemd in Azure: [Azure VM with systemd](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/no-agent)
[Automating running code on the1st boot] (https://docs.microsoft.com/en-us/azure/virtual-machines/linux/no-agent#automating-running-the-code-at-first-boot)

**Comment by aeschli** [comment]
The VSCode server currently behaves a bit different than a service/daemon: It's the client that installs and starts a VSCode server based on the client's build version.




For VSCode the server must match the the exact version of the client. So 

The `wslServer` script gets a exact version of the client as argument. The script will download the 
The server corresponding to that version will then be installed and ran.

**Comment by PavelSosin-320** [comment]
@aeschli It requires the reliable check that the "remote machine" is running and ready to serve inside the reasonable timeout period. Starting both WSL distro and VSCode server using a single command wsl -d distro ... genie -s with all functional dependencies like OCI engine, login daemon, journal daemon, network manager, session manager, etc. looks more reliable. 
Systemd-based WSL distros like the real Linux don't work correctly without regular user login because the plain wsl -u linux-user skips login daemon execution in Ubuntu and Fedora. User must enter username and password to create a "session".

**Comment by PavelSosin-320** [comment]
Yes, it's possible. I did it for Ubuntu and Fedora 33 that I use for Podman 3 RC testing. It works nice. It is absolutely necessary in the cases when software uses xxxxctl utilities for network handling, cgroup management, user session management, Linux "machine" management, etc. wsl -u doesn't offer the necessary capabilities to create Linux session allocate, and free resources for the user. It works similarly for the physical machine and VM. Since most of WSL 2 distros are plain repacked VM images of systemd-based Linux VM images and share the corresponding Linux package repositories even pure CLI utilities assume that native Linux systemd is available and all necessary services are managed by systemd,

**Comment by ssaavedra** [comment]
@aeschli there is more information and a walkaround at https://github.com/microsoft/vscode-remote-release/issues/5145
I believe this should now be on your backlog, but essentially a better solution would be that instead of launching `wsl.exe %USERPROFILE%\.vscode\extensions\wslServer.sh` you allowed to override the `wsl.exe` command-line to allow configuring it to `wsl.exe genie -c` before the `$(wslpath.exe "%USERPROFILE%\.vscode\extensions\wslServer.sh")`.

**Comment by bamurtaugh** [comment]
Thanks for opening! Closing in favor of https://github.com/microsoft/vscode-remote-release/issues/5145.
